pkg/kubelet/dockershim/docker_sandbox.go

```diff
@@ -124,36 +127,51 @@ func (ds *dockerService) StopPodSandbox(podSandboxID string) error {
 		namespace = m.Namespace
 		name = m.Name
 	} else {
-		checkpoint, err := ds.checkpointHandler.GetCheckpoint(podSandboxID)
-		if err != nil {
-			glog.Errorf("Failed to get checkpoint for sandbox %q: %v", podSandboxID, err)
-			return fmt.Errorf("failed to get sandbox status: %v", statusErr)
-		}
+		var checkpoint *PodSandboxCheckpoint
+		checkpoint, checkpointErr = ds.checkpointHandler.GetCheckpoint(podSandboxID)
```

Comment:
I don't understand why we still set namespace and name when `GetCheckpoint` already returns error.

Why don't we let `GetCheckpoint` return nil when there is an error?

Or is there any special consideration here?